### PR TITLE
Merge pull request #2 from OmarHamed0/login_feature_SCRUM-1

### PR DESCRIPTION
**Pull request by OmarHamed0**
add gust user

### CODE REVIEW

File: lib/src/presentation/managers/sign_in/sign_in_actions.dart [hunk 1]
```diff
@@ -3,3 +3,4 @@ sealed class SignInActions{}
 class SignInClickAction extends SignInActions{}
```
Comment (ahmed-sala) [inline lib/src/presentation/managers/sign_in/sign_in_actions.dart:3]: name it SigninAction better

File: lib/src/presentation/pages/sign_in/sign_in_screen_body.dart [hunk 2]
```diff
@@ -25,22 +25,24 @@ class SignInScreenBody extends StatelessWidget {
...
+            ErrorDialog.hide(context);
+          });
           ErrorDialog.show(context);
-           showToast(state.message!);
+          showToast(state.message!);
         } else if (state is PopDialogState) {
-           LoadingDialog.hide(context);
+          LoadingDialog.hide(context);
+        } else if (state is GuestLoginState) {
+          _goNextHome(context);
```
Comment (ahmed-sala) [inline lib/src/presentation/pages/sign_in/sign_in_screen_body.dart]: merge from development and then merge

**Review by ahmed-sala — CHANGES_REQUESTED** [review]
merge from dev

**Review by ahmed-sala — APPROVED** [review]

**Review by ali72-20 — APPROVED** [review]
all done

**Pull request merged**